lib/core/test/unit/Cardano/Wallet/Primitive/TypesSpec.hs

```diff
@@ -905,30 +917,5 @@ instance Arbitrary EpochLength where
     arbitrary = EpochLength . getNonZero <$> arbitrary
 
 instance Arbitrary SlotParameters where
-    arbitrary = SlotParameters <$> arbitrary <*> arbitrary <*> arbitrary
+    arbitrary = SlotParameters <$> arbitrary <*> arbitrary
     shrink = genericShrink
-
-instance {-# OVERLAPS #-} Arbitrary (SlotParameters, SlotId) where
-    arbitrary = do
-        (el, slot) <- arbitrary
-        sl <- arbitrary
-        st <- arbitrary
-        pure (SlotParameters el sl st, slot)
-    shrink (SlotParameters el sl st, slot) = do
-        (el', slot') <- shrink (el, slot)
-        pure (SlotParameters el' sl st, slot')
-
--- | Note, for functions which works with both an epoch length and a slot id,
--- we need to make sure that the 'slotNumber' doesn't exceed the epoch length,
--- otherwise, all computations get mixed up.
-instance {-# OVERLAPS #-} Arbitrary (EpochLength, SlotId) where
-    shrink (a,b) =
-        filter validSlotConfig $ zip (shrink a) (shrink b)
-      where
-        validSlotConfig (EpochLength ep, SlotId _ sl) = sl < ep
-
-    arbitrary = do
-        (EpochLength epochLength) <- arbitrary
-        ep <- choose (0, 1000)
-        sl <- choose (0, fromIntegral epochLength - 1)
-        return (EpochLength epochLength, SlotId ep sl)
```

Comment:
I particularly like the fact that we can now remove these overlapping instances.